workbaskets/views/ui.py

```diff
@@ -1086,7 +1086,10 @@ def get_form_kwargs(self):
         return kwargs
 
     def form_valid(self, form):
-        if not PackagedWorkBasket.objects.filter(workbasket=self.object).exists():
+        if (
+            not PackagedWorkBasket.objects.filter(workbasket=self.object).exists()
+            and not self.object.tasks.exists()
+        ):
             self.object.delete()
         else:
             self.object.archive()
```

Comment:
Now that a compound bool expression is required, I think this could be simplified.
```suggestion
        if (
           PackagedWorkBasket.objects.filter(workbasket=self.object).exists()
           or self.object.tasks.exists()
        ):
            self.object.archive()
        else:
            self.object.delete()
```